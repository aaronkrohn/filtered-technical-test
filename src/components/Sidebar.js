import React from 'react';
import styled from 'styled-components';
import AssetList from './AssetList';

const Wrapper = styled.div`
  background: #f8f8f8;
  flex: 0 0 33%;
  max-width: 33%;
  min-height: 0.063em;
  padding: 0 1em;
`;

const Sidebar = () => (
  <Wrapper>
    <AssetList />
  </Wrapper>
);

export default Sidebar;
