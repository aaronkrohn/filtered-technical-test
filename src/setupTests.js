import React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import reducers from './reducers/index';

export const renderWithRedux = (
  ui,
  { initialState, store = createStore(reducers, initialState) } = {}
) => ({
  ...render(<Provider store={store}>{ui}</Provider>),
  store
});
