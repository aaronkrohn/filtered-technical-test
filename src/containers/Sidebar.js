import { connect } from 'react-redux';
import Sidebar from '../components/Sidebar';

const mapStateToProps = ({ sidebarReducer }) => ({
  assetList: sidebarReducer.assetList
});

export default connect(mapStateToProps)(Sidebar);
