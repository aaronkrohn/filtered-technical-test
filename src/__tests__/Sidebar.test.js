import React from 'react';
import { render } from 'react-testing-library';
import { renderWithRedux } from '../setupTests';
import Sidebar from '../components/Sidebar';
import ConnectedSidebar from '../containers/Sidebar';
import assets from '../assets';

test('renders', () => {
  const { container } = render(<Sidebar />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders with Redux', () => {
  const { container } = renderWithRedux(<ConnectedSidebar />);
  const assetName = assets[0].name;
  expect(container.querySelector('h2')).toHaveTextContent(assetName);
});
