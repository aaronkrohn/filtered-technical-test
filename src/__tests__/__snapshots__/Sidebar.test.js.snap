// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  class="sc-ifAKCX dqEFoq"
>
  <ul
    class="sc-bxivhb bbZEIZ"
  >
    <li
      class="sc-bdVaJa iTWRwI"
      type="Article"
    >
      <h2
        class="sc-bwzfXH cvsGNX"
      >
        10 Principles of Change Management
      </h2>
      <p
        class="sc-htpNat tMhnM"
      >
        <span>
          0.11 min
        </span>
        <span>
          Article
        </span>
      </p>
    </li>
    <li
      class="sc-bdVaJa NSasu"
      type="Video"
    >
      <h2
        class="sc-bwzfXH cvsGNX"
      >
        Overview of change management phases
      </h2>
      <p
        class="sc-htpNat tMhnM"
      >
        <span>
          0.06 min
        </span>
        <span>
          Video
        </span>
      </p>
    </li>
    <li
      class="sc-bdVaJa hQXXxF"
      type="Document"
    >
      <h2
        class="sc-bwzfXH cvsGNX"
      >
        Blockchain for dummies
      </h2>
      <p
        class="sc-htpNat tMhnM"
      >
        <span>
          0.15 min
        </span>
        <span>
          Document
        </span>
      </p>
    </li>
    <li
      class="sc-bdVaJa hOixVw"
      type="Online Course"
    >
      <h2
        class="sc-bwzfXH cvsGNX"
      >
        10 Principles of Change Management
      </h2>
      <p
        class="sc-htpNat tMhnM"
      >
        <span>
          0.02 min
        </span>
        <span>
          Online Course
        </span>
      </p>
    </li>
    <li
      class="sc-bdVaJa NSasu"
      type="Video"
    >
      <h2
        class="sc-bwzfXH cvsGNX"
      >
        It's ok I'm only busy for a while
      </h2>
      <p
        class="sc-htpNat tMhnM"
      >
        <span>
          0.11 min
        </span>
        <span>
          Video
        </span>
      </p>
    </li>
    <li
      class="sc-bdVaJa hQXXxF"
      type="Document"
    >
      <h2
        class="sc-bwzfXH cvsGNX"
      >
        Why Carving Out 5 Hours Every Day For Difficult Work Can Lead To Greatness
      </h2>
      <p
        class="sc-htpNat tMhnM"
      >
        <span>
          0.09 min
        </span>
        <span>
          Document
        </span>
      </p>
    </li>
    <li
      class="sc-bdVaJa hQXXxF"
      type="Document"
    >
      <h2
        class="sc-bwzfXH cvsGNX"
      >
        Multi-tasking: you're making yourself slower and dumber
      </h2>
      <p
        class="sc-htpNat tMhnM"
      >
        <span>
          0.21 min
        </span>
        <span>
          Document
        </span>
      </p>
    </li>
    <li
      class="sc-bdVaJa NSasu"
      type="Video"
    >
      <h2
        class="sc-bwzfXH cvsGNX"
      >
        Overview of change management phases
      </h2>
      <p
        class="sc-htpNat tMhnM"
      >
        <span>
          0.06 min
        </span>
        <span>
          Video
        </span>
      </p>
    </li>
  </ul>
</div>
`;
